plugins/modules/ds8000_assign_host_port.py

```diff
@@ -0,0 +1,191 @@
+      - If set to False, in case that a host port is assigned to another host, it won't transfer it to the host that specified.
+    default: False
+    type: bool
+extends_documentation_fragment:
+  - ibm.ds8000.ds8000.documentation
+'''
+
+EXAMPLES = r'''
+- name: Assign some host ports to host with force on false
+  ibm.ds8000.ds8000_assign_host_port:
+    hostname: "{{ ds8000_host }}"
+    username: "{{ ds8000_username }}"
+    password: "{{ ds8000_password }}"
+    name: host_name_test
+    host_port:
+      - 10000000C9A1BAB2
+      - 10000000C9A1BAB3
+
+- name: Assign some host ports to host with force on true
+  ibm.ds8000.ds8000_assign_host_port:
+    hostname: "{{ ds8000_host }}"
+    username: "{{ ds8000_username }}"
+    password: "{{ ds8000_password }}"
+    name: host_name_test
+    host_port: 10000000C9A1BAB2
+'''
+
+RETURN = r''' # '''
+
+import json
+
+try:
+    import pyds8k.exceptions
+except ImportError:
+    pass
+
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils.ds8000 import (PyDs8k, costume_get_request,
+                                   ds8000_argument_spec)
+
+
+class PyDs8kHelper(PyDs8k):
+    def __init__(self, module):
+        super(PyDs8kHelper, self).__init__(module)
+        self.changed = False
+        self.failed = False
+
+    def _ds8000_assign_host_port(self, host_port):
+        result = {}
+        name = self.params['name']
+        changed, failed = self._assign_specific_host_port_to_specific_host(
+            host_port)
+        result = {'changed': changed, 'failed': failed}
+        return result
+
+    def _assign_specific_host_port_to_specific_host(self, host_port):
+        name = self.params['name']
+        client = self.client
+        if not self.params['force']:
+            check_if_wwpn_is_free, the_host_with_the_desired_host_port = self._check_if_the_host_port_is_free(
+                host_port)
+            if check_if_wwpn_is_free:
+                self.failed = True
+                self.module.fail_json(
+                    msg="The WWPN {host_port} assiened to another host ({host}) in the storage.\n"
+                    "For removing it from his current host and moving it to his desired host,"
+                    "use the force: true parameter.".format(
+                        host_port=host_port, host=the_host_with_the_desired_host_port))
+                return self.changed, self.failed
+        wwpns = self._get_all_the_host_ports_under_specific_host(name)
+        if host_port not in wwpns:
+            try:
+                client.update_host_port_change_host(
+                    port_id=host_port, host_name=name)
+                self.changed = True
+            except pyds8k.exceptions.BadRequest as bad_wwpn:
+                self.failed = True
+                if ':' in host_port:
```

Comment:
I suggest letting the user put `:` as popular in ports, but remove them before using them instead of making them do it themselves.